src/cli/cli.cpp

```diff
@@ -136,14 +155,51 @@ const struct Command Interpreter::sCommands[] =
     { "whitelist", &Interpreter::ProcessWhitelist },
 };
 
+#ifdef OTDLL
+uint32_t otPlatRandomGet(void)
+{
+    return (uint32_t)rand();
+}
+#else
+void otFreeMemory(const void *)
+{
+    // No-op on systems running OpenThread in-proc
+}
+#endif
+
+template <class T> class otPtr
```

Comment:
Would you really mind keeping it the way it is? It keeps the amount of code that is required to be maintains a little smaller, because only one typedef has to be defined for each type.